Hyperspace/Classes/Service/Network/NetworkActivityIndicatable.swift

```diff
@@ -0,0 +1,84 @@
+//
+//  NetworkActivityIndicatable.swift
+//  Hyperspace-iOS
+//
+//  Created by William McGinty on 12/21/17.
+//  Copyright © 2017 Bottle Rocket Studios. All rights reserved.
+//
+
+import Foundation
+
+/// Represents an object that can indicate to the user that network activity is currently taking place.
+public protocol NetworkActivityIndicatable {
+    var isNetworkActivityIndicatorVisible: Bool { get set }
+}
+
+/// Represents an object that is capable of recording the state of network requests and forwarding that state onto the corresponding indicator
+protocol NetworkActivityObservable {
+    func start()
+    func stop()
+    
+    var activityCount: Int { get }
+    var indicator: NetworkActivityIndicatable { get }
+}
+
+/// Manages and records the number of active network requests in relation to their effect on network indicators.
+public class NetworkActivityController: NetworkActivityObservable {
+    
+    // MARK: - Properties
+    public let delayInterval: TimeInterval
+    public private(set) var indicator: NetworkActivityIndicatable
+    
+    private let queue = DispatchQueue(label: "com.hyperspace.networkactivityindicator", qos: .userInitiated)
```

Comment:
Wondering if we should go ahead and expand out the label a bit more (not that it matters much). We'll want to think about how to standardize this across our projects too.

`private let queue = DispatchQueue(label: "com.bottlerocketstudios.hyperspace.networkactivityindicator", qos: .userInitiated)`

Reply:
I see no reason why not.